### DISCUSSION TITLE
Hide Relation column in response?

### DISCUSSION THREAD

**Discussion by ansyori28** [pocketbase/pocketbase#1648] [Q&A] [answered]
Hello, I'm interested in trying pocketbase for my next project, however a thought came to mind about a collection relationship feature in pocketbase. For example, I have a given collections/relationship setup below

<img width="667" alt="image" src="https://user-images.githubusercontent.com/8540792/213763161-5cf7fec1-f91b-4fff-b170-0123c556a17a.png">

When I fetch the data from api, pb give me this result:
<img width="630" alt="image" src="https://user-images.githubusercontent.com/8540792/213763365-1f16f973-2a4e-435a-9bb1-1f74a429e084.png">

As we can see from the 2nd screenshot, the 1st post item has a **comments** field which has all the _comment_id_ from the given comments I created, and also we have the full data of each comments inside the expand/comments response.

I was wondering, does the **comments** object that contains only array of string id necessary? can I only have one expand/comments response instead of two separate array of string/objects? because as my understanding, as the data grows, if for example if I have 50 of comments in one post, that would became 100 data in each 2 objects (50 in comments and other 50 in expand/comments object)

**Comment by ganigeorgiev** [comment, marked as answer]
The main record fields are preserved in order to have a consistent model representation.

Option to return only a subset of fields is in my todo and eventually will be implemented in the future but for now is a low priority (you could also check https://github.com/pocketbase/pocketbase/discussions/1605, https://github.com/pocketbase/pocketbase/discussions/354).
